docs/hazmat/primitives/asymmetric/rsa.rst

```diff
@@ -80,6 +80,37 @@ password. If the key is encrypted we can pass a ``bytes`` object as the
 There is also support for :func:`loading public keys in the SSH format
 <cryptography.hazmat.primitives.serialization.load_ssh_public_key>`.
 
+Key serialization
+~~~~~~~~~~~~~~~~~
+
+If you have a previously loaded or generated key that has the
+:class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPrivateKeyWithSerialization`
+interface you can use
+:meth:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPrivateKeyWithSerialization.dump`
+to serialize the key.
+
+.. doctest::
+
+    >>> from cryptography.hazmat.primitives import serialization
+    >>> pem = private_key.dump(
+    ...    serialization.PKCS8(serialization.Encoding.PEM),
```

Comment:
Yup, sounds good.

On Mon, Feb 23, 2015 at 6:23 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/hazmat/primitives/asymmetric/rsa.rst
> <https://github.com/pyca/cryptography/pull/1503#discussion_r25223831>:
>
> > @@ -80,6 +80,37 @@ password. If the key is encrypted we can pass a ``bytes`` object as the
> >  There is also support for :func:`loading public keys in the SSH format
> >  <cryptography.hazmat.primitives.serialization.load_ssh_public_key>`.
> >
> > +Key serialization
> > +~~~~~~~~~~~~~~~~~
> > +
> > +If you have a previously loaded or generated key that has the
> > +:class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPrivateKeyWithSerialization`
> > +interface you can use
> > +:meth:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPrivateKeyWithSerialization.dump`
> > +to serialize the key.
> > +
> > +.. doctest::
> > +
> > +    >>> from cryptography.hazmat.primitives import serialization
> > +    >>> pem = private_key.dump(
> > +    ...    serialization.PKCS8(serialization.Encoding.PEM),
>
> I'm fine with that. Should we consider having PKCS8/TraditionalOpenSSL be
> inside a Format enum then rather than being instances of empty classes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1503/files#r25223831>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084